proofs/tree_test.go

```diff
@@ -1281,11 +1245,57 @@ func Test_GenerateSingleLeafTree(t *testing.T) {
 	assert.Equal(t, foobarHash[:], doctree.RootHash())
 }
 
+func Test_SaltMessage(t *testing.T) {
+	doctree := NewDocumentTree(TreeOptions{Hash: sha256Hash})
+	err := doctree.AddLeavesFromDocument(&documentspb.ExampleContainSaltsDocument)
+	assert.Nil(t, err)
+	err = doctree.Generate()
+	assert.Nil(t, err)
+	assert.Len(t, doctree.leaves, 2)
+	assert.Equal(t,doctree.leaves[0].Salt, []byte{0x1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0x2}, "Salt should Match with the one contained in salt message")
+	assert.Equal(t,doctree.leaves[1].Salt, []byte{0x3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0x4}, "Salt should Match with the one contained in salt message")
+
+	doctree2 := NewDocumentTree(TreeOptions{Hash: sha256Hash})
+	document := &documentspb.ExampleContainSaltsDocument
+	document.Salts = nil
+	err = doctree2.AddLeavesFromDocument(document)
+	assert.Nil(t, err)
+	err = doctree2.Generate()
+	assert.Nil(t, err)
+	assert.Len(t, doctree2.leaves, 2)
+	for _, leaf := range doctree2.leaves {
+		assert.NotNil(t, leaf.Salt)
+	}
+}
+
+func Example_complete() {
+	// ContainSalts is a protobuf message which contain salts message
+	document := documentspb.ContainSalts{
```

Comment:
@zyfrank could you make sure this matches the example in the README?

Reply:
@lucasvo  I dont think `someMessage.Salts = []*proofspb.Salts{}` is needed . Take a look at `Test_ReturnGeneratedSalts`

```
	doc := new (documentspb.ContainSalts)
	doc.ValueA = "TestA"
	doc.ValueB = 5
	assert.Nil(t, doc.Salts)
	doctree := NewDocumentTree(TreeOptions{Hash: sha256Hash})
	err := doctree.AddLeavesFromDocument(doc)
	assert.Nil(t, err)
	assert.Equal(t, len(doc.Salts), 2)
```

Reply:
What I meant here is to make it use the same example document as in README.